software/src/applets/WTVCO.h

```diff
@@ -0,0 +1,691 @@
+            double t = static_cast<double>(i) / size;
+            double value = std::exp(-5 * t); // Exponential decay
+            waveform[i] = static_cast<int8_t>(127 * value);
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_ExponentialDecay2(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            waveform[i] = static_cast<int8_t>(127 * std::exp(-5 * t)); // Decays over the range
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_InverseExponential(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            double value = 1.0 - std::exp(-5 * t); // Inverse exponential growth
+            waveform[i] = static_cast<int8_t>(127 * value);
+        }
+        return waveform;
+    }
+
+
+
+// modified sinusoids
+
+    std::vector<int8_t> GenerateWaveForm_OffsetSine(size_t size, double offset = 0.5) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            waveform[i] = static_cast<int8_t>(127 * (std::sin(TWO_PI * t) + offset));
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_OffsetSine2(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            double value = 0.5 + 0.5 * std::sin(2 * M_PI * t); // Sine wave offset to [0, 1]
+            waveform[i] = static_cast<int8_t>(127 * value - 64); // Offset to [-64, 63]
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_Cosine(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            waveform[i] = static_cast<int8_t>(127 * std::cos(TWO_PI * t));
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_SinusoidalAM(size_t size) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            double value = std::sin(2 * M_PI * t) * (0.5 + 0.5 * std::sin(4 * M_PI * t)); // Sine modulated by sine
+            waveform[i] = static_cast<int8_t>(127 * value);
+        }
+        return waveform;
+    }
+
+    std::vector<int8_t> GenerateWaveForm_ClippedSine(size_t size, double clip = 0.5) {
+        std::vector<int8_t> waveform(size);
+        for (size_t i = 0; i < size; ++i) {
+            double t = static_cast<double>(i) / size;
+            double sineValue = std::sin(TWO_PI * t);
+            sineValue = std::clamp(sineValue, -clip, clip);
```

Comment:
For anything that won't just be compiled away, you probably want to prefer arduino functions over std functions. So here, I would use `constrain` over `clamp`.